server/data/data-file/src/main/java/org/apache/james/mailrepository/file/FileMailRepository.java

```diff
@@ -245,7 +299,42 @@ public Mail retrieve(MailKey key) throws MessagingException {
     }
 
     @Override
-    protected void internalRemove(MailKey key) throws MessagingException {
+    public void remove(Mail mail) throws MessagingException {
+        remove(MailKey.forMail(mail));
+    }
+
+    @Override
+    public void remove(Collection<Mail> mails) throws MessagingException {
+        for (Mail mail : mails) {
```

Comment:
mails.forEach?

Reply:
@blackheaven  can you contribute these fixes?